mantidproject/mantid
# Sample logs dialog highlight data with NeXus value alarms

**Do Not Merge Before #28564**

Shades in logs with invalid data in red, and a deeper red if all of the values are invalid.
Also hides (does not display) the invalid log filter logs.

**To test:**
1. Open Workbench
1. Load ENGINX00228061_log_alarm_data (unit test data, added recently)
1. right click ws->show sample logs
   ![image](https://user-images.githubusercontent.com/1017173/80803723-123ede80-8bab-11ea-92b4-35bbfc24f0ad.png)
1. cryo_temp1 (which has 1 invalid entry) should be shaded red
1. cryo_temp2 (which is all invalid) should be shaded deeper red
1. the tooltips should make sense
1. You should not see the logs cryo_temp1_invalid_values or cryo_temp2_invalid_values 


re #28569


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
